src/services/role.service.ts

```diff
@@ -1,23 +1,31 @@
+import { GuildMember } from 'discord.js';
 import { RoleType, RoleTypeKey } from './../common/types';
-import { GuildMember, Role, Collection } from 'discord.js';
 
+// roles are a hierarchical enum/type. the following is the hierarchy (use these keys defind in src/common/types in 'RoleType'):
+//   Admin
+//   Moderator
+//   Teaching Assistant
+//   RegularUser
+//   Suspended
+// an Admin can run everythin a Mod and below can run, a Mod can run everythin a TA and below can run ...
+// exception:
+// Suspended role supersedes any other role.
+// i.e., a member with both the with the Suspended and Admin roles can only run what a Suspended user can run
 export class RoleService {
-  hasPermission(member: GuildMember, minRoleToRun: number) {
-    const roles = member.roles;
-    const highestRole = this._getHighestRole(roles.cache);
-    return highestRole >= minRoleToRun;
-  }
-
-  private _getHighestRole(roles: Collection<string, Role>) {
-    let highestRole = 0;
-
-    roles.forEach((role: Role) => {
-      if (role.name in RoleType) {
-        if (role.name === 'Suspended') {return RoleType.Suspended;}
-        highestRole = Math.max(highestRole, RoleType[<RoleTypeKey>role.name]);
+  hasPermission(member: GuildMember, minRoleToRun: RoleType): boolean {
+    let memberRole: RoleType = RoleType.RegularUser;
+    const orderedRoleTypes: RoleTypeKey[] = [
+      'Suspended',
+      'Admin',
+      'Moderator',
```

Comment:
`Suspended` should be below `Admin` and `Moderator`

Sometime we do it as a joke or for whatever reason, and mod / admin should supersede suspended

Reply:
I can change this; I had heard differently from the implemention spec though.

Reply:
Just tested in production and it has no effect. So not really sure what was with the implementation before but I'd like it to stay that way. Where it has no effect